src/onemod/dtypes/column_spec.py

```diff
@@ -1,18 +1,39 @@
-from typing import List, Type
+from typing import Any, List, Type
 
 from pydantic import BaseModel, field_serializer, field_validator
 
 from onemod.constraints.base import Constraint
 
 
 class ColumnSpec(BaseModel):
-    type: Type[int] | Type[float] | Type[str] | Type[bool] | Type[BaseModel] | None = None
+    type: (
+        Type[int]
+        | Type[float]
+        | Type[str]
+        | Type[bool]
+        | Type[BaseModel]
+        | None
+    ) = None
     constraints: List[Constraint] | None = None
-    
+
+    def __getitem__(self, key: str) -> Any:
+        if key == "type":
+            return self.type
+        elif key == "constraints":
+            return self.constraints
+        else:
+            raise KeyError(f"Invalid key: {key}")
+
+    def __contains__(self, key: str) -> bool:
+        return key in {"type", "constraints"}
+
+    def keys(self):
+        return ["type", "constraints"]
```

Comment:
I might be really out of this, but curious to know why we want to make `ColumnSpec` dictionary like?

Reply:
There is a longer story to this where `ColumnSpec` started out as a `TypedDict` way back when.. But now the only reason really is because we are accessing it like a dictionary elsewhere in the package.